docs/reference/modules/discovery/discovery-settings.asciidoc

```diff
@@ -16,13 +23,13 @@ Discovery and cluster formation are affected by the following settings:
     to use to obtain the addresses of the seed nodes used to start the
     discovery process. By default, it is the
     <<settings-based-hosts-provider,settings-based seed hosts provider>>.
+    
+`discovery.type`::
 
-`cluster.initial_master_nodes`::
-
-    Sets a list of the <<node.name,node names>> or transport addresses of the
-    initial set of master-eligible nodes in a brand-new cluster. By default
-    this list is empty, meaning that this node expects to join a cluster that
-    has already been bootstrapped. See <<initial_master_nodes>>.
+    Specifies whether to use the default discovery and cluster formation (`zen`).
+    If set to `single-node`, the node elects itself master and does not join a
+    cluster with any other node. For more information about when you might use
+    this setting, see <<single-node-discovery>>.
 
```

Comment:
I'd like to avoid documenting the default being the literal `zen` to discourage this from appearing in any config files in the wild. Can we phrase it more as if the options are either to leave it unset or to set it to `single-node`?

Also it's not quite true that the node elects itself master if you set this setting. If it was previously part of a multi-node cluster then it may not be safe to restart in a single-node cluster like this, and we prevent this from happening.

Suggest something like this:

```suggestion
 
    Specifies whether {es} should form a multiple-node cluster or not. By
    default {es} will discover other nodes when forming a cluster and will allow
    other nodes to join the cluster later.  If `discovery.type` is set to
    `single-node` then {es} will form a single-node cluster. For more
    information about when you might use this setting, see
    <<single-node-discovery>>.
```